packages/cadl-ranch/src/actions/server-test.ts

```diff
@@ -0,0 +1,386 @@
+        helperCodeContent += `if(response.data.value !== undefined && Array.isArray(response.data.value)){
+          for(const val of response.data.value){
+            if(val["systemData"] !== undefined){
+              delete val["systemData"]["createdAt"];
+              delete val["systemData"]["lastModifiedAt"];
+            }
+          }
+          }`;
+      }
+    }
+    return helperCodeContent;
+  }
+
+  private getAssertContent(mockMethod: MockMethod): string {
+    let assertContent = `assert.strictEqual(response.status, ${mockMethod.response.status});`;
+    if (mockMethod.response.data !== undefined) {
+      if (
+        this.endpoint.endsWith(`response/custom-content-type`) ||
+        this.endpoint.endsWith(`response/octet-stream`) ||
+        (this.endpoint.endsWith(`/content-negotiation/different-body`) &&
+          mockMethod.request.config &&
+          mockMethod.request.config.headers &&
+          mockMethod.request.config.headers["accept"] === "application/json") ||
+        (mockMethod.request.config &&
+          mockMethod.request.config.headers &&
+          ["image/png", "image/jpeg"].includes(mockMethod.request.config.headers["accept"]))
+      ) {
+        assertContent += `assert.strictEqual(${mockMethod.response.data});`;
+      } else if (
+        mockMethod.response.data.contentType !== undefined &&
+        mockMethod.response.data.contentType === "application/xml"
+      ) {
+        assertContent += `assert.deepEqual(response.data, \`${mockMethod.response.data.rawContent}\`);`;
+      } else if (mockMethod.response.data.rawContent !== undefined) {
+        assertContent += `assert.deepEqual(JSON.stringify(response.data), ${JSON.stringify(mockMethod.response.data.rawContent)});`;
+      } else {
+        assertContent += `assert.deepEqual((response.data), ${JSON.stringify(mockMethod.response.data)});`;
+      }
+    }
+    if (mockMethod.response.headers !== undefined) {
+      assertContent += this.getResponseHeadersBlock(mockMethod.response.headers);
+    }
+    return assertContent;
+  }
+
+  private getTestCaseContent(mockMethod: MockMethod): string {
+    return `const endPoint = \`\${serverBasePath}${this.endpoint}\`;
+      ${this.getResponseContent(mockMethod)}
+      ${this.getHelperCodeContent(mockMethod)}
+      ${this.getAssertContent(mockMethod)}
+    `;
+  }
+
+  private getTestCasesContent(): string {
+    if (!this.mockMethods) return "";
+    let testCasesContent = "";
+    let counter = 0;
+    for (const mockMethod of this.mockMethods) {
+      testCasesContent += `it("Test Case ${counter + 1}", async () => {
+        ${this.getTestCaseContent(mockMethod)}
+      });`;
+      counter++;
+    }
+    return testCasesContent;
+  }
+
+  private getFileContents(): string {
+    return `${this.getImportContents()}\n${this.getGlobalDeclarationContents()}\n${this.getDescribeContents()}`;
+  }
+
+  public async generateFile() {
```

Comment:
This feels very weird to resort to that, why not directly run things, this is going to make debugging impossible. 
I don't feel like there is much point running vitest as well on top or any other test runner. We won't use the language for the test runner

Reply:
this just feels like a lot of pain to maintain regardless, now if we need to add some more complex scenario you don't need to just make the call yourself you need to reserialize that code to a file(the whole request creation that is currently hardcoded in that file).
What is the benefit of using a test runner here? I think it would be much better to just have the cli and if having vitest IDE support is important then you call to vitest api directly without shelkling a new process and writing new files.